File: android/app/src/main/java/com/antoninclouet/hanzilock/data/HanziCard.kt
package com.antoninclouet.hanzilock.data

/** Le hanzi sert d'identifiant unique dans ce MVP (pas de doublons dans le deck). */
data class HanziCard(
    val hanzi: String,
    val pinyin: String,
    val meaningFr: String,
    val hskLevel: Int,
    val category: String,
) {
    val id: String get() = hanzi
}

File: Shared/HanziCard.swift
import Foundation

struct HanziCard: Identifiable, Codable, Hashable {
    /// Le hanzi sert d'identifiant unique dans ce MVP (pas de doublons dans le deck).
    var id: String { hanzi }

    let hanzi: String
    let pinyin: String
    let meaningFR: String
    let hskLevel: Int
    let category: String
}
